staging/src/k8s.io/apiserver/pkg/storage/value/transformer.go

```diff
@@ -144,3 +144,78 @@ func (t *prefixTransformers) TransformToStorage(data []byte, context Context) ([
 	prefixedData = append(prefixedData, result...)
 	return prefixedData, nil
 }
+
+// LocationTransformer holds a transformer interface and the location of the resources it has to be used for.
```

Comment:
I had mentioned this in the description of the PR as well. I was also considering moving the PrefixTransformer and MutableTransformer to separate files. Let me know if that is required.